tests/runtime/core_internal_logger.c

```diff
@@ -0,0 +1,147 @@
+    struct mk_event_loop *evl       = NULL;
+
+    ret_ctx = flb_calloc(1, sizeof(struct http_client_ctx));
+    if (!TEST_CHECK(ret_ctx != NULL)) {
+        flb_errno();
+        TEST_MSG("flb_calloc(http_client_ctx) failed");
+        return NULL;
+    }
+
+    evl = mk_event_loop_create(16);
+    if (!TEST_CHECK(evl != NULL)) {
+        TEST_MSG("mk_event_loop failed");
+        flb_free(ret_ctx);
+        return NULL;
+    }
+    ret_ctx->evl = evl;
+    flb_engine_evl_init();
+    flb_engine_evl_set(evl);
+
+    ret_ctx->config = flb_config_init();
+    if (!TEST_CHECK(ret_ctx->config != NULL)) {
+        TEST_MSG("flb_config_init failed");
+        mk_event_loop_destroy(evl);
+        flb_free(ret_ctx);
+        return NULL;
+    }
+
+    ret_ctx->u = flb_upstream_create(ret_ctx->config, "127.0.0.1", 2020, 0, NULL);
+    if (!TEST_CHECK(ret_ctx->u != NULL)) {
+        TEST_MSG("flb_upstream_create failed");
+        flb_config_exit(ret_ctx->config);
+        mk_event_loop_destroy(evl);
+        flb_free(ret_ctx);
+        return NULL;
+    }
+
+    ret_ctx->u_conn = flb_upstream_conn_get(ret_ctx->u);
+    TEST_CHECK(ret_ctx->u_conn != NULL);
+
+    ret_ctx->u_conn->upstream = ret_ctx->u;
+
+    return ret_ctx;
+}
+
+int http_client_ctx_destroy(struct http_client_ctx *http_ctx) {
+    return flb_upstream_conn_release(http_ctx->u_conn);
+}
+
+/*
+ * Test that internal logs (i.e. those created by flb_info, flb_error, etc)
+ * tick internal v2 metrics.
+ */
+static void test_internal_log_metrics() {
+    flb_ctx_t *ctx;
+    int ret;
+    struct http_client_ctx *http_ctx;
+    struct flb_http_client *http_client;
+    size_t b_sent;
+    struct flb_regex *regex;
+
+    ctx = flb_create();
+    TEST_ASSERT(ctx != NULL);
+
+    TEST_ASSERT(flb_service_set(ctx,
+                                "HTTP_Server",
+                                "On",
+                                "HTTP_Listen",
+                                "127.0.0.1",
+                                "HTTP_Port",
+                                "2020",
+                                NULL) == 0);
+
+    ret = flb_start(ctx);
+    TEST_ASSERT(ret == 0);
+
+    http_ctx = http_client_ctx_create();
+    TEST_ASSERT(http_ctx != NULL);
+
+    http_client = flb_http_client(http_ctx->u_conn,
+                                  FLB_HTTP_GET,
+                                  "/api/v2/metrics/prometheus",
+                                  "", /* body */
+                                  0,  /* len(body) */
+                                  "127.0.0.1",
+                                  2020,
+                                  NULL,
+                                  0);
+    TEST_ASSERT(http_client != NULL);
+
+    /* The following does not update metrics since the test thread does not have an
+     * associated flb_worker */
+    flb_error("ignore this -- it's from the unit test itself");
+
+    /* TODO: this shouldn't rely on a startup delay */
```

Comment:
Is there a way to hook into the running flb `ctx` to wait until it's ready and serving all the http endpoints?

I don't want to sleep. I either want to block until the flb is ready or, less preferably, do the asserting in a loop like this:
```
while (elapsed time less than 5 seconds) {
	if (!call_http_and_check()) {
		sleep(50ms);
	}	
}
```